sdk/keyvault/keyvault-keys/migration-guide.md

```diff
@@ -0,0 +1,251 @@
+
+#### Authenticating
+
+Previously in `azure-keyvault` you could create a `KeyVaultClient` by using credentials from `ms-rest-azure` (up to the version `^2.6.0`. Higher versions are not supported).
+
+
+```js
+var KeyVault = require('azure-keyvault');
+var msRestAzure = require('ms-rest-azure');
+
+const clientId = "client id";
+const secret = "client secret";
+const domain = "tenant id";
+const vaultUrl = `https://my-vault.vault.azure.net/`;
+
+async function main() {
+  const credentials = await msRestAzure.loginWithServicePrincipalSecret(
+    clientId,
+    secret,
+    domain
+  );
+  const client = new KeyVault.KeyVaultClient(credentials);
+  const keyVaultKey = await client.getKey(vaultUrl, "MyKey", "");
+  console.log(keyVaultKey);
+}
+
+main().catch((err) => console.error(err));
+```
+
+Now in `@azure/keyvault-keys` you can create a `KeyClient` using any credential from [`@azure/identity`][identity-readme].
+
+You can install them by simply running the following command at the root of your project:
+
+```
+npm install --save @azure/identity @azure/keyvault-keys
+```
+
+Below is a simple example using both `@azure/keyvault-keys` and [`DefaultAzureCredential`][identity-readme-DAC]:
+
+```ts
+// The default credential first checks environment variables for configuration as described above.
+// If environment configuration is incomplete, it will try managed identity.
+const { KeyClient } = require("@azure/keyvault-keys");
+const { DefaultAzureCredential } = require("@azure/identity");
+
+// Azure SDK clients accept the credential as a parameter
+const credential = new DefaultAzureCredential();
+const vaultUrl = "https://my-vault.vault.azure.net/";
+const client = new KeyClient(vaultUrl, credential);
+
+async function main(): Promise<void> {
+  const keyVaultKey = await client.getKey("MyKey");
+  console.log(keyVaultKey);
+}
+
+main().catch((err) => console.error(err));
+```
+
+The new client also provides a `CryptographyClient` to perform cryptographic operations (encrypt/decrypt, wrap/unwrap, sign/verify) using a particular key. Below is an example.
+
+```ts
+const { KeyClient, CryptographyClient } = require("@azure/keyvault-keys");
+const { DefaultAzureCredential } = require("@azure/identity");
+
+const credential = new DefaultAzureCredential();
+const vaultUrl = "https://my-vault.vault.azure.net/";
+const client = new KeyClient(vaultUrl, credential);
+
+async function main(): Promise<void> {
+  const keyVaultKey = await client.getKey("MyKey");
+  const cryptographyClient = new CryptographyClient(keyVaultKey.id!, credential);
+  const signed = await cryptographyClient.signData("RS256", "some data");
+  console.log(signed);
+}
+
+main().catch((err) => console.error(err));
+```
+
+### Create a key
+
+In `azure-keyvault` you could create a key by using `KeyVaultClient`'s `createKey` method, which required a vault endpoint, key name, and key type. This method returned a `KeyBundle` containing the key.
```

Comment:
Mentioning just the return type does not help much. I would recommend to either skipping it or mention something interesting that can then be compared and contrasted with what we do in Track 2. If there is a major difference between `KeyBundle` and `KeyVaultKey`, then it would be worth having a section for that since most methods on the client deal with these types

Reply:
I think it's worth to mention the differences in the types, but that's going to take a while. I'll add a section with the differences on January.